scripts/fetch_smugmug_activity.py

```diff
@@ -0,0 +1,104 @@
+from datetime import datetime
+import re
+from xml.dom.minidom import Element
+import requests
+from xml.etree import ElementTree
+
+xml_namespaces = {'atom': 'http://www.w3.org/2005/Atom'}
+
+comment_feed_url = 'https://smugmug.com/hack/feed.mg?Type=usercomments&Data=cmubuggy&format=atom10'
+photo_feed_url = 'https://smugmug.com/hack/feed.mg?Type=nicknameRecent&Data=cmubuggy&format=atom10'
+
+
+# Run with 'python3 -m scripts.fetch_smugmug_activity'
+def main():
+    try:
+        recent_comments = fetch_recent_comments(comment_feed_url)
+        _print_items(recent_comments)
+    except:
+        pass
+
+    try:
+        recent_photos = fetch_recent_photos(photo_feed_url)
+        _print_items(recent_photos)
+    except:
+        pass
+
+    # TODO: Insert into database any new records
+    # TODO: Schedule with Crontab - https://towardsdatascience.com/how-to-schedule-python-scripts-with-cron-the-only-guide-youll-ever-need-deea2df63b4e
+    # TODO: Figure out where to route errors
+
+
+def fetch_recent_comments(url: str) -> 'list[dict]':
+    response = requests.get(url)
+    xml_root = ElementTree.fromstring(response.content)
+
+    recent_comments = []
+    for entry in _get_entries_from_xml_root(xml_root):
+        comment = {}
+        recent_comments.append(comment)
+
+        smugmug_id = _get_item_from_element(entry, 'id').text
+        matches = re.search(r'\Asmugmug:comment:(?P<comment_id>\d+)\Z', smugmug_id)
+        comment['comment_id'] = matches.groupdict()['comment_id']
+
+        title = _get_item_from_element(entry, 'title').text
+        matches = re.search(r'\A(?P<author>.*) commented on (.*) photo\Z', title)
+        comment['author'] = matches.groupdict()['author']
+
+        content = _get_item_from_element(entry, 'content').text
+        matches = re.search(
+          r'<img src="(?P<thumbnail_url>https:\/\/[^"]+\.[\w]+)".*>.*<p>Comment: (?P<comment>.*)<\/p>',
+          content)
+        matches_dict = matches.groupdict()
```

Comment:
Definitely - I currently having it skip the entire entry if any errors occur within that entry. Do you think we should allow for partial entry parsing and do exception catching at a lower level?

Reply:
That probably makes sense if there's some amount of data we can get by without actually having.  Otherwise, it seems better to just not let it into the database in the first place.  (So, possibly not skip it here, but certainly skip it before insert).